src/main/java/duke/task/TaskManager.java

```diff
@@ -0,0 +1,276 @@
+            + "%s\n"
+            + "Now you have left %d tasks in the list.";
+    private static final String COMPLETE_TASK_MESSAGE = "Good lad, you have finally completed the task you needed "
+            + "to do.\n"
+            + "%s";
+    private static final String ADD_TASK_MESSAGE = "Gaben have seen and will add the following task for you:\n"
+            + "%s\n"
+            + "You now have %d task in the list";
+    private static final String FILE_IO_ERROR_MESSAGE = "Sorry, there was some issues regarding the data files.";
+
+
+    /* List of tasks */
+    private final ArrayList<Task> tasksList;
+
+    /* Constructor for task manager */
+    public TaskManager() {
+        tasksList = new ArrayList<Task>();
+        Duke.printMessage("Gaben's Task Manager is here to assist you!");
+    }
+
+    /**
+     * Creates a new task based on type given
+     *
+     * @param userParameter Description of task and its parameter given by user
+     * @param type          type of task
+     * @return return task
+     * @throws InvalidParameterException If deadline/event task do not have the corresponding regex such as /by and /at
+     *                                   or when no description is given after the regex
+     */
+    public Task createTask(String userParameter, char type) throws InvalidParameterException {
+        Task task = null;
+        String[] userArguments = null;
+        boolean regexNotFound = false;
+        boolean emptyDescription = false;
+        switch (type) {
+        case CHAR_TYPE_TODO:
+            task = new Todo(userParameter);
+            break;
+        case CHAR_TYPE_EVENT:
+            userArguments = userParameter.split(EVENT_STRING_SPLIT_REGEX, 2);
+            regexNotFound = userArguments.length != 2;
+            emptyDescription = userArguments[0].isBlank();
+            if (emptyDescription || regexNotFound) {
+                throw new InvalidParameterException(EVENT_ERROR_MESSAGE);
+            }
+            task = new Event(userArguments[0], userArguments[1]);
+            break;
+        case CHAR_TYPE_DEADLINE:
+            userArguments = userParameter.split(DEADLINE_STRING_SPLIT_REGEX, 2);
+            regexNotFound = userArguments.length != 2;
+            emptyDescription = userArguments[0].isBlank();
+            if (emptyDescription || regexNotFound) {
+                throw new InvalidParameterException(DEADLINE_ERROR_MESSAGE);
+            }
+            task = new Deadline(userArguments[0], userArguments[1]);
+            break;
+        }
+        return task;
+    }
+
+
+    /**
+     * Adds a newly created task with given task description and type to tasks list
+     *
+     * @param userInputArray Command and parameters given by user
+     * @param type           Type of task given by user input
+     */
+    public void addTask(String[] userInputArray, TaskType type) {
+
+        // Check for empty description and return if no description
```

Comment:
Since the code logic is quite simple and easily understandable, you may consider removing these comments :)